sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -591,7 +591,8 @@ object FoldablePropagation extends Rule[LogicalPlan] {
     } else {
       CleanupAliases(plan.transformUp {
         // We can only propagate foldables for a subset of unary nodes.
-        case u: UnaryNode if foldableMap.nonEmpty && canPropagateFoldables(u) =>
+        case u: UnaryNode if foldableMap.nonEmpty && canPropagateFoldables(u)
```

Comment:
shall we update `canPropagateFoldables` instead?

Reply:
I reverted this part and changed `ResolveRecursiveReferneces` to use new attributes.